src/sciline/task_graph.py

```diff
@@ -79,3 +110,39 @@ def visualize(
         from .visualize import to_graphviz
 
         return to_graphviz(self._graph, **kwargs)
+
+    def _repr_html_(self) -> str:
+        leafs = sorted(
+            [
+                escape(keyname(key))
+                for key in (
+                    self._keys if isinstance(self._keys, tuple) else [self._keys]
+                )
+            ]
+        )
+        roots = sorted(
+            {
+                escape(keyname(key))
+                for key, (_, requires) in self._graph.items()
+                if len(requires) == 0
```

Comment:
Hm I'm not sure I understand. Is there a conceptual difference between a parameter and a provider without inputs?

Reply:
Yes, a parameter is typically something a user sets, i.e., it would typically be different every time the pipeline is used (and therefore seeing its value is valuable). A provider is more static and would typically not change (and we cannot get its value without calling it, which may be expensive).